model/src/main/kotlin/config/FileStorageConfiguration.kt

```diff
@@ -42,25 +42,22 @@ data class FileStorageConfiguration(
     /**
      * Create a [FileStorage] based on this configuration.
      */
-    fun createFileStorage(): FileStorage {
-        require((httpFileStorage == null) xor (localFileStorage == null)) {
-            "Exactly one implementation must be configured for a FileStorage."
-        }
+    fun createFileStorage(): FileStorage =
+        when {
```

Comment:
For me the first three lines of the old function helped quite a bit in terms of readability as it made it very obvious
that excatly one of the storages must be configured. The change IMO sacrifices that, which is why I'd prefer to fix that by just replacing `!!` with `checkNotNull`.

Reply:
I was actually trying to "kill two birds with one stone" here and also get rid of the IMO rather unreadable `let` / `?:` construct in this case. I'll see if I can come up with something that makes the "either or" check again performed earlier.

Reply:
I found a solution that I actually believe to be quite elegant, please have a look.